testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentModule.java

```diff
@@ -86,6 +85,22 @@ public EnvironmentConfig provideEnvironmentConfig(EnvironmentOptions options, En
         return factory.getConfig(options.config);
     }
 
+    @Provides
+    @Singleton
+    public PortBinder providePortBinder(EnvironmentOptions options)
+    {
+        PortBinder binder = new PortBinder.DefaultPortBinder();
+        if (options.bindPorts) {
+            binder = new PortBinder.FixedPortBinder();
+        }
+
+        if (options.bindPortsBase > 0) {
+            binder = new PortBinder.ShiftingPortBinder(binder, options.bindPortsBase);
```

Comment:
It does not make sense to shift when `DefaultPortBinder` is used. So I would verify that `options.bindPorts` is on.

Even better would be to remove `--no-bind` and `--bind-ports-base`. Make `--no-bind` behavior to be default. Add `--bind=<port_shift>` option where `<port_shift>` would default to 0.

WDYT?

Reply:
Done.

So there are 3 states:

default (no `--bind` param)  do not bind
`--bind` - bind ports shifted by 0 (so container port is mapped to host port)
`--bind=N` - bind ports shifted by N (so container port is mapped to host port + N)